add docs for isolated store per user docs (#4549)
langchain-ai/docs: src/langsmith/resource-auth.mdx

Reviewer: Should we remove this from here as well? Otherwise should change `last one` below back to the specific handler language you had.

Author: let's keep original example unless you think it should definitely be removed

Reviewer: Good with me! Suggestion to update last one to last two then.

        detail="User lacks the required permissions.",
    )

@auth.on.store
async def scope_store(ctx: Auth.types.AuthContext, value: Auth.types.on.store.value):
    """Isolate store data per user by rewriting namespaces."""
    namespace = tuple(value["namespace"]) if value.get("namespace") else ()
    if not namespace or namespace[0] != ctx.user.identity:
        namespace = (ctx.user.identity, *namespace)

>     value["namespace"] = namespace

Reviewer: I actually think the old example is still a good default too. I actually think probably good to show both and discuss the tradeoffs of the implementation (explicit namespace + deny vs. assumed user prefixing of the namespace). Maybe better on the store specific page though? Don't know that this handler even really needs to be a part of this page.

2. Reading threads
3. Accessing assistants

> The first three of these match specific **actions** on each resource (see [resource actions](/langsmith/auth#resource-specific-handlers)), while the last one (`@auth.on.assistants`) matches _any_ action on the `assistants` resource. For each request, LangGraph will run the most specific handler that matches the resource and action being accessed. This means that these handlers will run rather than the broadly scoped "`@auth.on`" handler.

Author: ```suggestion
he first three of these match specific **actions** on each resource (see [resource actions](/langsmith/auth#resource-specific-handlers)), while the last one (`@auth.on.assistants`) matches _any_ action on the `assistants` resource. For each request, LangGraph will run the most specific handler that matches the resource and action being accessed. This means that the four handlers above will run rather than the broadly scoped "`@auth.on`" handler.
```

> he first three of these match specific **actions** on each resource (see [resource actions](/langsmith/auth#resource-specific-handlers)), while the last one (`@auth.on.assistants`) matches _any_ action on the `assistants` resource. For each request, LangGraph will run the most specific handler that matches the resource and action being accessed. This means that the four handlers above will run rather than the broadly scoped "`@auth.on`" handler.

Author: ```suggestion
The first three of these match specific **actions** on each resource (see [resource actions](/langsmith/auth#resource-specific-handlers)), while the last one (`@auth.on.assistants`) matches _any_ action on the `assistants` resource. For each request, LangGraph will run the most specific handler that matches the resource and action being accessed. This means that the four handlers above will run rather than the broadly scoped "`@auth.on`" handler.
```

> The first three of these match specific **actions** on each resource (see [resource actions](/langsmith/auth#resource-specific-handlers)), while the last one (`@auth.on.assistants`) matches _any_ action on the `assistants` resource. For each request, LangGraph will run the most specific handler that matches the resource and action being accessed. This means that the four handlers above will run rather than the broadly scoped "`@auth.on`" handler.

Reviewer: ```suggestion
The first three of these match specific **actions** on each resource (see [resource actions](/langsmith/auth#resource-specific-handlers)), while the last two (`@auth.on.assistants` and `@auth.on.store`) matche _any_ action on the `assistants` and `store` resources. For each request, LangGraph will run the most specific handler that matches the resource and action being accessed. This means that the four handlers above will run rather than the broadly scoped "`@auth.on`" handler.
```
